pkg/sdk/testint/authentication_policies_gen_integration_test.go

```diff
@@ -0,0 +1,330 @@
+		err := client.AuthenticationPolicies.Create(ctx, req)
+		t.Cleanup(cleanupAuthenticationPolicyProvider(req.GetName()))
+
+		alterErr := client.AuthenticationPolicies.Alter(ctx, sdk.NewAlterAuthenticationPolicyRequest(req.GetName()).
+			WithSet(*sdk.NewAuthenticationPolicySetRequest().WithClientTypes([]sdk.ClientTypes{{ClientType: "DRIVERS"}, {ClientType: "SNOWSQL"}})))
+		require.NoError(t, alterErr)
+
+		desc, err := client.AuthenticationPolicies.Describe(ctx, req.GetName())
+		require.NoError(t, err)
+		assert.Contains(t, desc, sdk.AuthenticationPolicyDescription{Property: "CLIENT_TYPES", Value: "[DRIVERS, SNOWSQL]"})
+	})
+
+	t.Run("Alter - set security integrations", func(t *testing.T) {
+		secId := createSAML2Integration(t, func(r *sdk.CreateSaml2SecurityIntegrationRequest) {
+			r.WithEnabled(true)
+		})
+		req := defaultCreateRequest()
+		err := client.AuthenticationPolicies.Create(ctx, req)
+		t.Cleanup(cleanupAuthenticationPolicyProvider(req.GetName()))
+
+		alterErr := client.AuthenticationPolicies.Alter(ctx, sdk.NewAlterAuthenticationPolicyRequest(req.GetName()).
+			WithSet(*sdk.NewAuthenticationPolicySetRequest().WithSecurityIntegrations([]sdk.SecurityIntegrationsOption{{Name: secId.Name()}})))
+		require.NoError(t, alterErr)
+
+		desc, err := client.AuthenticationPolicies.Describe(ctx, req.GetName())
+		require.NoError(t, err)
+		assert.Contains(t, desc, sdk.AuthenticationPolicyDescription{Property: "SECURITY_INTEGRATIONS", Value: fmt.Sprintf("[%s]", secId.Name())})
+	})
+
+	t.Run("Alter - set mfa authentication methods", func(t *testing.T) {
+		req := defaultCreateRequest()
+		err := client.AuthenticationPolicies.Create(ctx, req)
+		t.Cleanup(cleanupAuthenticationPolicyProvider(req.GetName()))
+
+		alterErr := client.AuthenticationPolicies.Alter(ctx, sdk.NewAlterAuthenticationPolicyRequest(req.GetName()).
+			WithSet(*sdk.NewAuthenticationPolicySetRequest().WithMfaAuthenticationMethods([]sdk.MfaAuthenticationMethods{{Method: "PASSWORD"}})))
+		require.NoError(t, alterErr)
+
+		desc, err := client.AuthenticationPolicies.Describe(ctx, req.GetName())
+		require.NoError(t, err)
+		assert.Contains(t, desc, sdk.AuthenticationPolicyDescription{Property: "MFA_AUTHENTICATION_METHODS", Value: "[PASSWORD]"})
+	})
+
+	t.Run("Alter - set mfa enrollment", func(t *testing.T) {
+		req := defaultCreateRequest()
+		err := client.AuthenticationPolicies.Create(ctx, req)
+		t.Cleanup(cleanupAuthenticationPolicyProvider(req.GetName()))
+
+		alterErr := client.AuthenticationPolicies.Alter(ctx, sdk.NewAlterAuthenticationPolicyRequest(req.GetName()).
+			WithSet(*sdk.NewAuthenticationPolicySetRequest().WithMfaEnrollment("REQUIRED")))
+		require.NoError(t, alterErr)
+
+		desc, err := client.AuthenticationPolicies.Describe(ctx, req.GetName())
+		require.NoError(t, err)
+		assert.Contains(t, desc, sdk.AuthenticationPolicyDescription{Property: "MFA_ENROLLMENT", Value: "REQUIRED"})
+	})
+
+	t.Run("Alter - set comment", func(t *testing.T) {
```

Comment:
@sfc-gh-jcieslak do we need such granular tests here?

Reply:
Addressed in https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3068 I merged those tests into one `set and unset`